extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpAuthenticator.java

```diff
@@ -137,10 +137,14 @@ public Uni<Boolean> sendChallenge(RoutingContext routingContext) {
         routingContext.request().resume();
         Uni<Boolean> result = null;
 
-        HttpAuthenticationMechanism matchingMech = routingContext.get(HttpAuthenticationMechanism.class.getName());
-        if (matchingMech != null) {
-            result = matchingMech.sendChallenge(routingContext);
+        // only take mechanism from context if it is path specific as we still want to send challenge to a custom mechanism
+        if (usePathSpecificMechanism(routingContext)) {
```

Comment:
See my earlier comment, I think it should check if more than one mechanism is reg-ed - if it is only one then no need to check the mech in the context

Reply:
I think you are right, I'll try it.